### PR TITLE
19085-Translation-in-tier-price-phtml-not-working

### PR DESCRIPTION
**Pull request by agorbulin**
#19085:Translation in tier_price.phtml not working

### Description (#19085 )
provided fix uses PHP translate instead of not working JS translate

### Fixed Issues (if relevant)
1. magento/magento2#19085:Translation in tier_price.phtml not working 

### Manual testing scenarios (*)
1. Configure tier prices for configurable product (for his simples)
2. Change translates for phrases in Magento/ConfigurableProduct/i18n/en_US.csv:
"Buy %1 for %2 each and","Buy %1 for %2 each and"
save,save
3. Flush caches
4. Go to PDP
5. Choose option of configurable with tier prices
6. Check translates

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @agorbulin. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — CHANGES_REQUESTED** [review]
> A possible workaround for this issue is to use `<?php echo __() ?>` instead but i don't think this is a proper solution for this issue.

Makes sense to me. So, @agorbulin, why JS translation is not working in this case in the first place?

**Comment by agorbulin** [comment]
@orlangur 
JS translation is not working in phtml files, because only JS files and HTML templates are parsed during generation js-translation.json, see here
 https://github.com/magento/magento2/blob/d7012b2ca1495676478d8afed8bf016ccd03bc8a/app/code/Magento/Translation/Model/Js/DataProvider.php#L99
So, there are 2 solutions:
- use PHP translates in phtml (I chose it)
- move embed js to own js file and use js translation ($.mage.__())

File: app/code/Magento/ConfigurableProduct/view/base/templates/product/price/tier_price.phtml [hunk 1]
```diff
@@ -15,9 +15,9 @@
                 + '</span>'
             + '</span>'; %>
         <li class="item">
-            <%= $t('Buy %1 for %2 each and').replace('%1', item.qty).replace('%2', priceStr) %>
+            <%= $t('<?= $block->escapeHtml(__('Buy %1 for %2 each and')) ?>').replace('%1', item.qty).replace('%2', priceStr) %>
```
Comment (orlangur) [inline app/code/Magento/ConfigurableProduct/view/base/templates/product/price/tier_price.phtml]: Why JS translation `$t` is still here?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/ConfigurableProduct/view/base/templates/product/price/tier_price.phtml (hunk 1 above)
Comment (agorbulin) [reply app/code/Magento/ConfigurableProduct/view/base/templates/product/price/tier_price.phtml]: Extra `$t` removed, please check.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3546](https://jira.corp.magento.com/browse/ENGCOM-3546) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @agorbulin. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**